### ISSUE TITLE
Draw Decals over Items

### ISSUE THREAD

**Issue by hubismal** [Regalis11/Barotrauma#3824]
[Feature Request, Design, Code]
I'm requesting drawing decals over items with kinematic physics bodies beyond the player depth of 0.5.
I feel like intentionally not drawing decals over items was probably a design choice, but I'm asking for reconsideration:

- **Having decals cover up items is no longer a huge issue now that things can be cleaned**. If anything, things being dirty will actually be disadvantageous as it obscures labels, vents, items, etc, and therefore cleaning will have a purpose (keep things visible and usable) while still being a downtime activity (it's extremely unlikely things would get dirty enough to warrant cleaning as high priority).
- **Our new "artists" will hardly have any canvas space**. 90% of custom subs use every inch of free wall space they can find to fill up with scaled down items. Since you can only draw on clean walls (or at least only have your art visible on clean walls) there will be some subs where there's just no room to draw or view decals on at all
- **There would be more evidence of chaos left behind from explosions and bloodstains**. One of the nice new things about the decal system is that there's evidence left behind when someone is shot or something explodes and so on. Except if you get hurt or something explodes in front of an item, the decal is left behind the item, which means that the whole concept is kind of squandered because the evidence will always be behind items.

tl;dr it would give cleaning a purpose, make art a viable downtime even on subs where no wall is visible, and make killers have to clean up after themselves.

I'll even go out on a limb and say that cleaning and painting's current implementation are pretty boring when they're basically just the cosmetic wallpaper of the sub.

**Comment by hubismal** [comment]
Rokvach previously suggested that internal testing was done and that having 2d decals everywhere over everything else made the subs look flat. This could probably be averted by preserving the "shading" (the net low/dark RGB) for items that are covered with decals, if such a thing is possible.
![image](https://user-images.githubusercontent.com/47284081/94275652-bba9b680-ff36-11ea-839e-d4fc5946b478.png)
The red part of the picture above was achieved by converting the sprites to greyscale and converting the white to alpha.

**Comment by ezjamsen** [comment]
I have no problems with this, in theory. @EdusFF what do you think?
